File: 3550-smallest-index-with-digit-sum-equal-to-index/3550-smallest-index-with-digit-sum-equal-to-index.py
class Solution:
    def digitSum(self,n):
        sum =0
        while n:
            digit = n % 10
            sum += digit
            n = n // 10
        return sum
    def smallestIndex(self, nums: List[int]) -> int:
        n = len(nums)
        ans = -1
        for i in range(n):
            if self.digitSum(nums[i])==i:
                ans =i
                break
        return ans

File: 3550-smallest-index-with-digit-sum-equal-to-index/3550-smallest-index-with-digit-sum-equal-to-index.cpp
class Solution {
public:
    int digitSum(int n){
        int sum =0;
        while(n){
            int digit = n % 10;
            sum +=digit;
            n = n/10;
        }
        return sum;
    }
    int smallestIndex(vector<int>& nums) {
        
        int n = nums.size();
        int ans =-1;

        for(int i =0;i<n;i++){
            if(digitSum(nums[i])==i){
                ans = i;
                break;
            }
        }
        return ans;
    }
};